### PR TITLE
GET users endpoint

### PR DESCRIPTION
**Pull request by mattpeyton**
GET method that serves an array of all users.

### CODE REVIEW

**Comment by bschuste** [comment]
Looks good to me

**Review by bschuste — APPROVED** [review]
We get the whole list of users info, might be useful to be able to get the info about one user.

File: api/swagger/swagger.yaml [hunk 1]
```diff
@@ -18,25 +18,21 @@ consumes:
...
+      parameters: []
       responses:
         "200":
           description: Success
           schema:
-            # a pointer to a definition
-            $ref: "#/definitions/HelloWorldResponse"
+            type: array
+            items:
+              $ref: "#/definitions/User"
```
Comment (bretwalker) [inline api/swagger/swagger.yaml]: The objects being returned do not match the schema defined by `#/definitions/User`.

Here's what Swagger UI says the response should look like:
```
[
  {
    "firstName": "string",
    "lastName": "string",
    "birthMonth": 0,
    "birthDay": 0,
    "birthYear": 0,
    "email": "string"
  }
]
```

Here's an actual response:
```
[
  {
    "id": "1",
    "firstName": "Jane",
    "lastName": "Smith",
    "birthDate": "1980-01-27",
    "email": "jane.smith@example.com"
  }
]
```

You'll need to manipulate the object you get back from the database to match the format defined in Swagger.

**Review by bretwalker — CHANGES_REQUESTED** [review]

**Comment by bretwalker** [comment]
@bschuste Yes, we should add `GET /users/{user id}` to our list of endpoints for sure.

We also need to update our user models and endpoints to support the new fields @cjrobles12 added https://github.com/codelouisvilleproject/database-schema/commit/36663dedc0b890061ae20fc3c45209210abd71a0. It would be easier to do this sooner rather than later to reduce the number of pieces of code that need to be updated.

File: api/swagger/swagger.yaml (hunk 1 above)
Comment (bretwalker) [reply api/swagger/swagger.yaml]: You can use composition to inherit from the `User` definition. You'll want to have all of its fields in your objects, and you'll want to add an the ID. You could call your definition `UserWithId`.

https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#models-with-composition

**Comment by mattpeyton** [comment]
Thanks for pointing out the discrepancy, Bret.  I updated the swagger.yaml and the controller to match each other and made sure that the response matched the story that I'm working from.  Any future changes we would like to make to the response array (adding or removing attributes) should be pretty straightforward.

**Comment by bretwalker** [comment]
I actually think it would be best if the PUT/GET were as similar as possible.
Did you see my note about using composition to inherit from Brigitte's definition? You should be able to inherit from her `User` definition and then you only need to add an ID field, since her definition contains the other fields already.
This way, it reduces code, makes our API more consistent, and lets us make updates in one place.
If you need help with this, just let me know.
Thanks!

**Comment by mattpeyton** [comment]
Bret. Sure. I've just now changed the swagger.yaml to inherit Brigitte's definition and add an id. Before I go any further, do I need the result to match the GET/user endpoint story or should we update the story to show a more detailed array (full birthdate + id)?

**Review by bretwalker — APPROVED** [review]
Looks great!

**Pull request merged**